lib/pure/nativesockets.nim

```diff
@@ -374,6 +374,19 @@ proc getHostByName*(name: string): Hostent {.tags: [ReadIOEffect].} =
   result.addrList = cstringArrayToSeq(s.h_addr_list)
   result.length = int(s.h_length)
 
+proc getHostname*(): string {.tags: [ReadIOEffect].} =
+  ## Returns the local hostname (not the FQDN)
+  const size = 64
+  var hostname = cstring(newString(size))
```

Comment:
Do it this way instead:

```nim
result = newString(size)
... pass result to getHostname
let x = len(cstring(result))
result.setLen(x)
```

Your way is more expensive (2 allocations instead of one) and dangerous (GC could remove the cstring under the hood).